### ISSUE TITLE
Prop : canDrop - function does not define nextParent

### ISSUE THREAD

**Issue by Phawaq** [nosferatu500/react-sortable-tree#8]
Hi, 
first of all I would like to say that you have done a great job.

I found a small bug in the Props **canDrop** this function does not define the **nextParent** parameter, it is always **undefined**

## expected input
```
{
	node: object,  
	prevPath: number[] or string[],  
	prevParent: object,  
	prevTreeIndex: number,  
	nextPath: number[] or string[],  
	nextParent: object,  
 	nextTreeIndex: number  
} 
```

## obtained input 
![image](https://user-images.githubusercontent.com/9798940/142450650-dbd251b5-197f-40b9-b9c3-fbc93e57fbc3.png)

## Others

I have also check that I don't have access to external variables or states of the component's parent.

### Example declaration in the parent
```
const [parentCheck, setParentCheck] = useState(false);

const eventCanDropNode = (args: any, parentCheck:boolean, otherExternVariable:Object) => {        
        console.log(parentCheck); //return undefined
        console.log(otherExternVariable); //return undefined

        if (parentCheck) {
            return false;
        }                
        return true;
    }

return (<SortableTree
	treeData={nodes}
	onChange={treeData => onchangeTreeData(treeData)}
	canDrop={args => eventCanDropNode(args, parentCheck,otherExternVariable )}
	...
/>)

```

**Comment by nosferatu500** [comment]
Thanks for reporting this! I will check it out in the coming days.

**Comment by harrylepotter-win** [comment]
Thanks for the amazing work @nosferatu500 . I'm also having the similar issues. `nextParent` and `nextPath`  seems to be incorrect on the `canDrop` call:
![candroperror](https://user-images.githubusercontent.com/82542425/146721472-76956995-afa9-46aa-8daa-af2857ba75f6.png)

**Comment by harrylepotter-win** [comment]
update: i've isolated this bug down to `src/utils/tree-data-utils::addNodeAtDepthAndIndex`. Still trying to wrap my head around this block

**Comment by wsz7777** [comment]
have any progress @harrylepotter-win  ?

**Comment by harrylepotter-win** [comment]
@wsz7777 @nosferatu500 the following patch appears to make this work as expected. Tested against the 'Prevent drop' storybook:

```
index c475b00..268b0c5 100644
--- a/src/utils/dnd-manager.ts
+++ b/src/utils/dnd-manager.ts
@@ -183,23 +183,15 @@ const canDrop = (
 
   if (typeof treeRefcanDrop === 'function') {
     const { node } = monitor.getItem()
-    const addedResult = memoizedInsertNode({
-      treeData: draggingTreeData || treeReftreeData,
-      newNode: node,
-      depth: targetDepth,
-      getNodeKey,
-      minimumTreeIndex: dropTargetProps.listIndex,
-      expandParent: true,
-    })
 
     return treeRefcanDrop({
       node,
       prevPath: monitor.getItem().path,
       prevParent: monitor.getItem().parentNode,
       prevTreeIndex: monitor.getItem().treeIndex, // Equals -1 when dragged from external tree
-      nextPath: addedResult.path,
-      nextParent: addedResult.parentNode,
-      nextTreeIndex: addedResult.treeIndex,
+      nextPath: dropTargetProps.children.props.path,
+      nextParent: dropTargetProps.children.props.parentNode,
+      nextTreeIndex: dropTargetProps.children.props.treeIndex,
     })
   }
```

**Comment by nosferatu500** [comment]
The patch will be released with version 4.0.3.

@harrylepotter-win Thanks for this patch!

**Comment by wsz7777** [comment]
good!!!
